pfcpiface/messages.go

```diff
@@ -632,3 +666,73 @@ func (pc *PFCPConn) generateAssociationRequest(n4LocalIP string, n3ip string, n4
 		log.Fatalln("Unable to transmit association setup request ", err)
 	}
 }
+
+func readReportNotification(rn <-chan uint64, pfcpConn *PFCPConn,
+	udpConn *net.UDPConn, udpAddr net.Addr) {
+	log.Println("read report notification start")
+	for {
+		select {
+		case fseid := <-rn:
+			handleDigestReport(fseid, pfcpConn, udpConn, udpAddr)
+
+		default:
+			time.Sleep(2 * time.Second)
+		}
+	}
+}
+
+func handleDigestReport(fseid uint64,
+	pfcpConn *PFCPConn,
+	udpConn *net.UDPConn,
+	udpAddr net.Addr) {
+	session, ok := pfcpConn.mgr.sessions[fseid]
+	if !ok {
+		log.Println("No session found for fseid : ", fseid)
+		return
+	}
+
+	/* Number of Outstanding Notifies per session is 1 */
+	if session.notifyCount != 0 {
+		return
+	}
+
+	session.notifyCount++
+	pfcpConn.mgr.sessions[fseid] = session
+	seq := pfcpConn.getSeqNum()
+	serep := message.NewSessionReportRequest(0, /* MO?? <-- what's this */
+		0,                            /* FO <-- what's this? */
+		0,                            /* seid */
+		seq,                          /* seq # */
+		0,                            /* priority */
+		ie.NewReportType(0, 0, 0, 1), /*upir, erir, usar, dldr int*/
+	)
+	serep.Header.SEID = session.remoteSEID
+	var pdrID uint32
+	for _, pdr := range session.pdrs {
+		if pdr.srcIface == core {
+			pdrID = pdr.pdrID
+			break
+		}
+	}
+
+	log.Println("Pdr iD : ", pdrID)
+	if pdrID == 0 {
+		log.Println("No Pdr found for downlink")
+		return
+	}
```

Comment:
Should probably get this from fastpath along with FSEID, so that we are accurate.

Reply:
https://github.com/omec-project/upf-epc/issues/277

Have raised new issue to track this. Will fix as part of that.